src/units.cpp

```diff
@@ -460,6 +460,16 @@ std::string Units::unitAttributeReference(size_t index) const
     return ref;
 }
 
+void Units::setUnitAttributeReference(size_t index, const std::string &reference)
+{
+    UnitDefinition ud;
+    if (index < pFunc()->mUnitDefinitions.size()) {
+        ud = pFunc()->mUnitDefinitions.at(index);
+        ud.mReference = reference;
+        pFunc()->mUnitDefinitions[index] = ud;
+    }
+}
```

Comment:
Why not simply have:
```c++
if (index < pFunc()->mUnitDefinitions.size()) {
    auto ud = pFunc()->mUnitDefinitions.at(index);
    ud.mReference = reference;
    pFunc()->mUnitDefinitions[index] = ud;
}
```
or, at least:
```c++
if (index < pFunc()->mUnitDefinitions.size()) {
    UnitDefinition ud = pFunc()->mUnitDefinitions.at(index);
    ud.mReference = reference;
    pFunc()->mUnitDefinitions[index] = ud;
}
```

Reply:
[1901ad2](https://github.com/cellml/libcellml/pull/1167/commits/1901ad210ddf891f670bdd53ad0302c37078e132)